components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java

```diff
@@ -0,0 +1,655 @@
+        Mockito.when(this.deviceAccessAuthorizationService.isUserAuthorized(Mockito.any(DeviceIdentifier.class)))
+                .thenReturn(true);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getEventStreamAdminServiceStub"))
+                .toReturn(this.eventStreamAdminServiceStub);
+        Mockito.when(this.eventStreamAdminServiceStub.getStreamDefinitionDto(Mockito.anyString())).thenReturn(null);
+
+        Map<String, Object> payload = new HashMap<>();
+        CacheImpl cache = Mockito.mock(CacheImpl.class);
+
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getDynamicEventCache"))
+                .toReturn(cache);
+
+        Response response = this.deviceAgentService.publishEvents(payload, TEST_DEVICE_TYPE, TEST_DEVICE_IDENTIFIER);
+        Assert.assertNotNull(response, "Response should not be null");
+        Assert.assertEquals(response.getStatus(), Response.Status.BAD_REQUEST.getStatusCode(),
+                "The response status should be 400");
+        Mockito.reset(eventStreamAdminServiceStub);
+    }
+
+    @Test(description ="Test the error scenario of Publishing Events with null event attributes.")
+    public void testEventPublishWithEventAttributesNULLAndPublishEventsFailure() throws
+            DeviceAccessAuthorizationException, RemoteException {
+        PowerMockito.stub(PowerMockito.method(PrivilegedCarbonContext.class, "getThreadLocalCarbonContext"))
+                .toReturn(this.privilegedCarbonContext);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class,
+                "getDeviceAccessAuthorizationService")).toReturn(this.deviceAccessAuthorizationService);
+        Mockito.when(this.deviceAccessAuthorizationService.isUserAuthorized(Mockito.any(DeviceIdentifier.class)))
+                .thenReturn(true);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getEventStreamAdminServiceStub"))
+                .toReturn(this.eventStreamAdminServiceStub);
+        EventStreamAttributeDto eventStreamAttributeDto = Mockito.mock(EventStreamAttributeDto.class,
+                Mockito.RETURNS_MOCKS);
+        EventStreamDefinitionDto eventStreamDefinitionDto = Mockito.mock(EventStreamDefinitionDto.class,
+                Mockito.RETURNS_MOCKS);
+        Mockito.when(this.eventStreamAdminServiceStub.getStreamDefinitionDto(Mockito.anyString()))
+                .thenReturn(eventStreamDefinitionDto);
+        Mockito.when(eventStreamDefinitionDto.getPayloadData()).thenReturn(new EventStreamAttributeDto[]{});
+        EventsPublisherService eventPublisherService = Mockito.mock(EventsPublisherServiceImpl.class,
+                Mockito.RETURNS_MOCKS);
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getEventPublisherService")).toReturn
+                (eventPublisherService);
+
+        Map<String, Object> payload = new HashMap<>();
+        CacheImpl cache = Mockito.mock(CacheImpl.class);
+
```

Comment:
Remove the un-wanted lines. Please correct in other places as well

Reply:
This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e